kirby/modules.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Kirby;
+
+use A;
+use F;
+use Dir;
+use Response;
+
+class Modules {
+
+  // core
+  public $controllers = array();
+  public $languages   = array();
+  public $models      = array();
+  public $snippets    = array();
+  public $tags        = array();
+  public $templates   = array();
+  public $autocss     = array();
+  public $autojs      = array();
+
+  // panel
+  public $blueprints  = array();
+  public $fields      = array();
+
+
+  /**
+   * Magic getter for module arrays
+   */
+  public function __call($method, $arguments) {
+    return isset($this->{$method}) ? $this->{$method} : null;
+  }
+
+  /**
+   * Registers a plugin directory for a module type
+   */
+  public function register($module, $dir) {
+    if(!isset($this->{$module})) return false;
+
+    // prepare for auto-assets
+    if($module === 'autocss' or $module === 'autojs') {
+      $route = 'assets/modules/' . sha1($dir);
+      $dir   = array($dir, $route);
+      $this->assets($route, $dir);
+    }
+
+    array_push($this->{$module}, $dir);
+    return true;
+  }
+
+  /**
+   * Creates route for plugin assets
+   */
+   public function assets($route, $root) {
+    if($page = page($route)) return false;
+
+    return kirby()->routes(array(
+      array(
+        'pattern' => trim($route, '/') . '/(:all)',
+        'action'  => function($file) use($route, $root) {
+          $path = $root . DS . $file;
+          if(is_file($path)) {
+            $extension = substr(strchr($path, '.'), 1);
+            return new Response(f::read($path), $extension);
```

Comment:
Nope, throws an error for the css in my tests:
<img width="1004" alt="screen shot 2016-03-04 at 00 25 22" src="https://cloud.githubusercontent.com/assets/3788865/13513205/a6a857aa-e19f-11e5-9d35-f7988a46fc0e.png">